bundles/org.openhab.binding.openthermgateway/src/main/java/org/openhab/binding/openthermgateway/internal/OpenThermGatewaySocketConnector.java

```diff
@@ -0,0 +1,232 @@
+import java.net.Socket;
+import java.util.AbstractMap;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.types.State;
+import org.openhab.binding.openthermgateway.handler.TypeConverter;
+
+/**
+ * The {@link OpenThermGatewaySocketConnector} is responsible for handling the socket connection
+ *
+ * @author Arjen Korevaar - Initial contribution
+ * @author Arjan Mels - Improved robustness by re-sending commands, handling all message types (not only Boiler)
+ */
+@NonNullByDefault
+public class OpenThermGatewaySocketConnector implements OpenThermGatewayConnector {
+    private static final int COMMAND_RESPONSE_TIME = 100;
+    private static final int COMMAND_TIMEOUT = 5000;
+
+    private OpenThermGatewayCallback callback;
+    private String ipaddress;
+    private int port;
+
+    private @Nullable Socket socket;
+    private @Nullable BufferedReader reader;
+    private @Nullable PrintWriter writer;
+
+    private volatile boolean stopping;
+    private boolean connected;
+
+    public OpenThermGatewaySocketConnector(OpenThermGatewayCallback callback, String ipaddress, int port) {
+        this.callback = callback;
+        this.ipaddress = ipaddress;
+        this.port = port;
+    }
+
+    @Override
+    public void run() {
+        stopping = false;
+        connected = false;
+
+        try {
+            callback.log(LogLevel.Info,
+                    String.format("Connecting OpenThermGatewaySocketConnector to %s:%s", this.ipaddress, this.port));
+
+            callback.connecting();
+
+            socket = new Socket();
+            socket.connect(new InetSocketAddress(this.ipaddress, this.port), COMMAND_TIMEOUT);
+            socket.setSoTimeout(COMMAND_TIMEOUT);
+            writer = new PrintWriter(socket.getOutputStream(), true);
+            reader = new BufferedReader(new InputStreamReader(socket.getInputStream()));
+
+            connected = true;
+
+            callback.connected();
+
+            callback.log(LogLevel.Debug, "OpenThermGatewaySocketConnector connected");
+
+            sendCommand(GatewayCommand.parse(GatewayCommandCode.PrintReport, "A"));
+            // Set the OTGW to report every message it receives and transmits
+            sendCommand(GatewayCommand.parse(GatewayCommandCode.PrintSummary, "0"));
+
+            while (!stopping && !Thread.currentThread().isInterrupted()) {
+                @Nullable
+                String message = reader.readLine();
+
+                if (message != null) {
+                    handleMessage(message);
+                } else {
+                    callback.log(LogLevel.Info, "Connection closed by OpenTherm Gateway");
+                    break;
+                }
+            }
+
+            callback.log(LogLevel.Debug, "Stopping OpenThermGatewaySocketConnector");
+
+        } catch (Exception e) {
```

Comment:
Can you catch the specific exception here or add `RuntimException`
```suggestion
        } catch (Exception e) {
```

Reply:
Removed catching generic exceptions